R/readWrite.R

```diff
@@ -30,23 +30,43 @@ writeBambuOutput <- function(se, path, prefix = "") {
             "extended_annotations.gtf", sep = "")
         gtf <- writeToGTF(annotation = transcript_grList,
             file = transcript_gtffn)
-        transcript_counts <- as.data.frame(assays(se)$counts)
-        geneIDs <- data.frame(mcols(transcript_grList,
-            use.names = FALSE)$TXNAME, mcols(transcript_grList,
-            use.names = FALSE)$GENEID)
-        colnames(geneIDs) <- c("TXNAME", "GENEID")
-        transcript_counts <- cbind(geneIDs, transcript_counts)
-        transcript_countsfn <- paste(outdir, prefix, 
-            "counts_transcript.txt", sep = "")
-        utils::write.table(transcript_counts, file = transcript_countsfn,
-            sep = "\t", quote = FALSE, row.names = FALSE)
-        gene_se <- transcriptToGeneExpression(se)
-        gene_counts <- as.data.frame(assays(gene_se)$counts)
-        gene_countsfn <- paste(outdir, prefix, "counts_gene.txt", sep = "")
-        utils::write.table(gene_counts, file = gene_countsfn, 
-            sep = "\t", quote = FALSE)
+        
+        report <- data.table(v = c("counts","fullLengthCounts",
+                      "partialLengthCounts","uniqueCounts","counts"),
```

Comment:
this would be more robust if it just lists the names of the assays for v `assays(se)`. Gene can be done separately, does not need to be in the loop